### DISCUSSION TITLE
Using RTK Query for slow computations

### DISCUSSION THREAD

**Discussion by StefanBRas** [reduxjs/redux-toolkit#1898] [General]
I found this article called ["You can use React Query for slow computation, not just API"](https://swizec.com/blog/you-can-use-react-query-for-slow-computation-not-just-api/) interesting and was wondering if the same can be done in RTKQ?

I haven't looked much into it, but as understand it this can be done by using queryFn.

And if it is doable, it it then actually smart? Or is there better ways for these kind of tasks.

**Comment by phryneas** [comment]
Yes, that's perfectly fine. There are also many people using it for stuff like interfacing with asynchronous functions, like interacting with the OS in Expo etc.

Just make sure the result is serializable.

**Reply by StefanBRas** [reply]
Neat! Would it make sense to have an example with something like that in the docs?

Or perhaps it's an really obvious use case and I just missed it.

**Reply by phryneas** [reply]
Feel free to do a PR to the docs, it would be very welcome. Unfortunately I don't really have the time for it at the moment.
